File: src/C++/STL/0014.Longest Common Prefix 最长公共前缀.cpp
class Solution {    //最长公共前缀---STL
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";
        int minn = 0x3f3f3f3f;
        for(auto i:strs){   //找到其中最短长度的string
            if(i.size() < minn) minn = i.size();
        }
        string com; //构造后为 ""
        for(int j = 0;j < minn;j++){
            vector<string>::iterator i = strs.begin();  
            char c = (*i)[j];
            while(i < strs.end()){  //遍历,判断是非具有相同的c 
                if(c == (*i)[j]) i++;
                else return com;    //字符不同,返回结果
            }
            com += c;   //累加前缀
        }
        return com;
    }
};

File: src/Python/STL/0014.Longest Common Prefix 最长公共前缀.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        else:
            minn = min(strs)
            maxn = max(strs)
            for i,element in enumerate(minn):
                if(element != maxn[i]):
                    return minn[:i]
            return minn
